pkg/metrics/metrics.go

```diff
@@ -0,0 +1,104 @@
+package metrics
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+var (
+	OCMAgentTotalRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ocm_agent_total_requests",
+			Help: "A count of total requests to ocm agent service",
+		}, []string{})
+
+	OCMAgentByServiceRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ocm_agent_service_requests",
+			Help: "A count of total requests to ocm agent based on sub service",
+		}, []string{"path"})
+
+	OCMAgentTotalFailedRequests = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "ocm_agent_failed_requests",
+			Help: "A count of total failed request to ocm agent service",
+		}, []string{})
+
+	OCMAgentRequestFailure = prometheus.NewGaugeVec(
+		prometheus.GaugeOpts{
+			Name: "ocm_agent_request_failure",
+			Help: "Indicate that the request send to ocm agent failed",
+		}, []string{"path"})
```

Comment:
Could you please explain the difference between these?

Reply:
The `OCMAgentRequestFailure` is a gauge value, to indicate that the request to the ocm-agent is failed, and we are going to use this to alert SRE to fix the issue. The value will be change back to 0 once the issue is gone.
OCMAgentTotalFailedRequests is a counter, like statistics. To calculate the total failed request, can be also exported to external tools.